src/Common/src/Interop/Windows/IpHlpApi/Interop.NetworkInformation.cs

```diff
@@ -0,0 +1,585 @@
+        {
+            SkipUnicast = 0x0001,
+            SkipAnycast = 0x0002,
+            SkipMulticast = 0x0004,
+            SkipDnsServer = 0x0008,
+            IncludePrefix = 0x0010,
+            SkipFriendlyName = 0x0020,
+            IncludeWins = 0x0040,
+            IncludeGateways = 0x0080,
+            IncludeAllInterfaces = 0x0100,
+            IncludeAllCompartments = 0x0200,
+            IncludeTunnelBindingOrder = 0x0400,
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct IpSocketAddress
+        {
+            internal IntPtr address;
+            internal int addressLength;
+
+            internal IPAddress MarshalIPAddress()
+            {
+                // Determine the address family used to create the IPAddress.
+                AddressFamily family = (addressLength > Internals.SocketAddress.IPv4AddressSize)
+                    ? AddressFamily.InterNetworkV6 : AddressFamily.InterNetwork;
+                Internals.SocketAddress sockAddress = new Internals.SocketAddress(family, addressLength);
+                Marshal.Copy(address, sockAddress.Buffer, 0, addressLength);
+
+                return sockAddress.GetIPAddress();
+            }
+        }
+
+        // IP_ADAPTER_ANYCAST_ADDRESS
+        // IP_ADAPTER_MULTICAST_ADDRESS
+        // IP_ADAPTER_DNS_SERVER_ADDRESS
+        // IP_ADAPTER_WINS_SERVER_ADDRESS
+        // IP_ADAPTER_GATEWAY_ADDRESS
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct IpAdapterAddress
+        {
+            internal uint length;
+            internal AdapterAddressFlags flags;
+            internal IntPtr next;
+            internal IpSocketAddress address;
+
+            internal static InternalIPAddressCollection MarshalIpAddressCollection(IntPtr ptr)
+            {
+                InternalIPAddressCollection addressList = new InternalIPAddressCollection();
+
+                while (ptr != IntPtr.Zero)
+                {
+                    IpAdapterAddress addressStructure = Marshal.PtrToStructure<IpAdapterAddress>(ptr);
+                    IPAddress address = addressStructure.address.MarshalIPAddress();
+                    addressList.InternalAdd(address);
+                    ptr = addressStructure.next;
+                }
+
+                return addressList;
+            }
+
+            internal static IPAddressInformationCollection MarshalIpAddressInformationCollection(IntPtr ptr)
+            {
+                IPAddressInformationCollection addressList = new IPAddressInformationCollection();
+
+                while (ptr != IntPtr.Zero)
+                {
+                    IpAdapterAddress addressStructure = Marshal.PtrToStructure<IpAdapterAddress>(ptr);
+                    IPAddress address = addressStructure.address.MarshalIPAddress();
+                    addressList.InternalAdd(new SystemIPAddressInformation(address, addressStructure.flags));
+
+                    ptr = addressStructure.next;
```

Comment:
Nit: there's a blank line above this line but not above the corresponding one in the previous function.